### ISSUE TITLE
Tree 控件 onNodeExpand 没有触发

### ISSUE THREAD

**Issue by cwxyz007** [element-plus/element-plus#714]
示例仓库：https://github.com/cwxyz007/element-plus-test

如下代码中，当展开 Tree 控件节点的时候，没有触发 `onNodeExpand` 事件。

`onNodeCollapse` 能事件正确触发。

```tsx
export const App = defineComponent(() => {
    const data = [xxxx];

    return () => (
        <div style={{ width: '400px', height: '800px' }}>
            <el-tree
                data={data}
                onNodeCollapse={() => console.log('collapse')}
                onNodeExpand={() => console.log('expand')}
            ></el-tree>
        </div>
    );
});
```

**Comment by cwxyz007** [comment]
https://github.com/element-plus/element-plus/blob/53dfce0903dba7620bb6d77a9d1c770beabb9ed0/packages/tree/src/tree-node.vue#L192-L201

看了一下源码，可能是这里的 `ctx` 不对。希望能帮到 debug。

**Comment by JeremyWuuuuu** [comment]
This might caused by Vue instead of ElementPlus see #678 

One way to verify this is to lock the version of Vue to 3.0.0

**Comment by JeremyWuuuuu** [comment]
We are expecting the fix from Vue

**Comment by cwxyz007** [comment]
> This might caused by Vue instead of ElementPlus see #678
> 
> One way to verify this is to lock the version of Vue to 3.0.0

I tried this way, But It's still not working.

And more, Both `onNodeExpand` and `onNodeCollapse` are not triggered.

**Comment by JeremyWuuuuu** [comment]
> > This might caused by Vue instead of ElementPlus see #678
> > One way to verify this is to lock the version of Vue to 3.0.0
> 
> I tried this way, But It's still not working.
> 
> And more, Both `onNodeExpand` and `onNodeCollapse` are not triggered.

Alright, then I'm going to find out what happened,thanks for bringing in this bug

**Comment by cwxyz007** [comment]
I have update the example, and it seem a bug of JSX compiler.

When I use `.vue` file, It works fine, but when I move to `.tsx`, it doesn't work.

**Comment by JeremyWuuuuu** [comment]
> I have update the example, and it seem a bug of JSX compiler.
> 
> When I use `.vue` file, It works fine, but when I move to `.tsx`, it doesn't work.

Because in the `.tsx` files, it was hand written, which does not match the event name `Vue` emits. In this repo, the compiler version was locked in `3.0.0`, which in your repo it's `^3.0.0`, these 2 were incompatible, so the bug occurs.

No worries, we are planning on bumping the version to `^3.0.3` today, so it should work fine after the bumped release.

**Comment by JeremyWuuuuu** [comment]
making fix in #729

**Comment by JeremyWuuuuu** [comment]
Fixed in #729
